addons/mrp/models/stock_move.py

```diff
@@ -274,6 +274,12 @@ def _onchange_product_uom_qty(self):
             new_qty = float_round((mo.qty_producing - mo.qty_produced) * self.unit_factor, precision_rounding=self.product_uom.rounding)
             self.quantity = new_qty
 
+    @api.onchange('quantity')
+    def _onchange_quantity(self):
```

Comment:
I suppose you can't fix this if the user set the qty through an rpc ?

Reply:
I guess that we can try to modify the compute method instead !